src/com/unbounce/dogstatsd/ring.clj

```diff
@@ -0,0 +1,105 @@
+(ns com.unbounce.dogstatsd.ring
+  "Ring middleware for instrumenting HTTP request/responses.
+
+  The middleware provides the following dogstatsd metrics:
+
+  - http.1xx  - counter of 1xx responses
+  - http.2xx  - counter of 2xx responses
+  - http.3xx  - counter of 3xx responses
+  - http.4xx  - counter of 4xx responses
+  - http.5xx  - counter of 5xx responses
+  - http.count     - counter for total requests
+  - http.exception - counter for exceptions raised
+  - http.duration  - histogram of request duration
+
+  See `wrap-ring-metric` for more information.
+  "
+  (:require [com.unbounce.dogstatsd.core :as statsd])
+  (:import [java.util.concurrent ThreadLocalRandom]))
+
+(defn sample? [sample-rate]
```

Comment:
since this overrides the dd-agent sample-rate feature, should we dissoc :sample-rate to avoid double applying the chance of skip sampling at statsd/increment and histogram API calls?

Reply:
Let's wait to hear back from the datadog team to get clarity how sample-rate works.

Reply:
As discussed (in chat), the app has to do some conditional sending to make use of `sample-rate`